Source/RestKit/JSONValue.swift

```diff
@@ -0,0 +1,148 @@
+            let description = "Failed to decode a JSON value from the given single value container."
+            let context = DecodingError.Context(codingPath: container.codingPath, debugDescription: description)
+            throw DecodingError.dataCorrupted(context)
+        }
+    }
+
+    /// Initialize a JSON value from an encodable type.
+    public init<T: Encodable>(from value: T) throws {
+        let encoder = JSONEncoder()
+        let decoder = JSONDecoder()
+        let data = try encoder.encode(value)
+        self = try decoder.decode(JSONValue.self, from: data)
+    }
+
+    /// Convert this JSON value to a decodable type.
+    public func toValue<T: Decodable>(_ type: T.Type) throws -> T {
+        let encoder = JSONEncoder()
+        let decoder = JSONDecoder()
+        let data = try encoder.encode(self)
+        return try decoder.decode(T.self, from: data)
+    }
+
+    /// Encode a JSON value.
+    public func encode(to encoder: Encoder) throws {
+        switch self {
+        case .null:
+            var container = encoder.singleValueContainer()
+            try container.encodeNil()
+        case .boolean(let boolean):
+            var container = encoder.singleValueContainer()
+            try container.encode(boolean)
+        case .string(let string):
+            var container = encoder.singleValueContainer()
+            try container.encode(string)
+        case .int(let int):
+            var container = encoder.singleValueContainer()
+            try container.encode(int)
+        case .double(let double):
+            var container = encoder.singleValueContainer()
+            try container.encode(double)
+        case .array(let array):
+            var container = encoder.unkeyedContainer()
+            try array.forEach() { try container.encode($0) }
+        case .object(let object):
+            var container = encoder.container(keyedBy: DynamicKeys.self)
+            try object.forEach() { key, value in
+                guard let codingKey = DynamicKeys(stringValue: key) else {
+                    let description = "Cannot construct CodingKey for \(key)"
+                    let context = EncodingError.Context(codingPath: encoder.codingPath, debugDescription: description)
+                    throw EncodingError.invalidValue(key, context)
+                }
+                try container.encode(value, forKey: codingKey)
+            }
+        }
+    }
+
+    /// Compare two JSON values for equality.
+    public static func == (lhs: JSONValue, rhs: JSONValue) -> Bool {
+        switch (lhs, rhs) {
+        case (.null, null): return true
+        case (.boolean(let x), .boolean(let y)): return x == y
+        case (.string(let x), .string(let y)): return x == y
+        case (.int(let x), .int(let y)): return x == y
+        case (.double(let x), .double(let y)): return x == y
+        case (.array(let x), .array(let y)): return x == y
+        case (.object(let x), .object(let y)): return x == y
```

Comment:
Do you have a test that shows that one `[String: JSONValue]` `==` another `[String: JSONValue]`?  I don't see and obvious one, and I think it would be worth making sure it's covered (both for the `object` and `array` case).

Reply:
Good catch! Just pushed up a test for `==`.